iofog-agent-daemon/src/test/java/org/eclipse/iofog/field_agent/FieldAgentTest.java

```diff
@@ -0,0 +1,1605 @@
+                .add("token", "token")
+                .add("message", "success").build();
+
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        Field instance = FieldAgent.class.getDeclaredField("instance");
+        instance.setAccessible(true);
+        instance.set(null, null);
+        MODULE_NAME = null;
+        Mockito.reset(fieldAgent, fieldAgentStatus, orchestrator);
+        if (method != null)
+            method.setAccessible(false);
+    }
+    /**
+     * Set a mock to the {@link FieldAgent} instance
+     * Throws {@link RuntimeException} in case if reflection failed, see a {@link Field#set(Object, Object)} method description.
+     * @param mock the mock to be inserted to a class
+     */
+    private void setMock(FieldAgent mock) {
+        try {
+            Field instance = FieldAgent.class.getDeclaredField("instance");
+            instance.setAccessible(true);
+            instance.set(instance, mock);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void initiateMockStart() {
+        thread = mock(Thread.class);
+        try {
+            whenNew(Thread.class).withParameterTypes(Runnable.class,String.class).withArguments(any(Runnable.class),
+                    anyString()).thenReturn(thread);
+            PowerMockito.doNothing().when(thread).start();
+            fieldAgent.start();
+        } catch (Exception e) {
+            fail("this should not happen");
+        }
+
+    }
+
+    /**
+     * Test module index of FieldAgent
+     */
+    @Test
+    public void testGetModuleIndex() {
+        assertEquals(5, fieldAgent.getModuleIndex());
+    }
+
+    /**
+     * Test Module Name of FieldAgent
+     */
+    @Test
+    public void getModuleName() {
+        assertEquals(MODULE_NAME, fieldAgent.getModuleName());
+
+    }
+
+    /**
+     * Test getInstance is same as mock
+     */
+    @Test
+    public void testGetInstanceIsSameAsMock() {
+        assertSame(fieldAgent, FieldAgent.getInstance());
+    }
+
+    /**
+     * Test postTracking with valid jsonObject
+     */
+    @Test ( timeout = 5000L )
+    public void testPostTrackingWithValidJsonObject() {
+        try {
+            initiateMockStart();
+            fieldAgent.postTracking(jsonObject);
+            Mockito.verify(orchestrator).request(any(), any(), any(), any());
+            PowerMockito.verifyStatic(LoggingService.class, Mockito.never());
+            LoggingService.logError(eq(MODULE_NAME), eq("Unable send tracking logs"), any());
+        } catch (Exception e) {
+            fail("This should not happen");
+        }
+    }
+
+    /**
+     * Test postTracking with null jsonObject
+     */
+    @Test ( timeout = 5000L )
+    public void throwsExceptionPostTrackingWithNullJsonObject() {
+        try {
+            initiateMockStart();
+            PowerMockito.when(orchestrator.request(any(), any(), any(), any())).thenThrow(mock(Exception.class));
```

Comment:
Looking at `orchestrator.request`, it creates an empty json object if provided parameter is null. I think it would be better to name test case `postTrackingLogsErrorWhenRequestFails`

Reply:
updated the name of method and verification